exo/main.py

```diff
@@ -78,6 +80,10 @@
   discovery = UDPDiscovery(args.node_id, args.node_port, args.listen_port, args.broadcast_port, lambda peer_id, address, device_capabilities: GRPCPeerHandle(peer_id, address, device_capabilities), discovery_timeout=args.discovery_timeout)
 elif args.discovery_module == "tailscale":
   discovery = TailscaleDiscovery(args.node_id, args.node_port, lambda peer_id, address, device_capabilities: GRPCPeerHandle(peer_id, address, device_capabilities), discovery_timeout=args.discovery_timeout, tailscale_api_key=args.tailscale_api_key, tailnet=args.tailnet_name)
+elif args.discovery_module == "manual":
+  if not args.discovery_config_path:
+    raise ValueError(f"--discovery-config-path is required when using manual discovery. Please provide a path to a config json file.")
```

Comment:
Here, and throughout this PR I raise exceptions when something in the config isn't correct. I don't see a lot of other situations where errors are being thrown back to the user elsewhere in the codebase, but I'm not sure how else to handle this.

How do you think these sorts of errors should be handled? One alternative is to just ignore them and assume no peers.

Reply:
This is good.